pkg/route/registry/route/strategy.go

```diff
@@ -43,6 +43,8 @@ func (routeStrategy) NamespaceScoped() bool {
 func (s routeStrategy) PrepareForCreate(obj runtime.Object) {
 	route := obj.(*api.Route)
 	route.Status = api.RouteStatus{}
+	// Do not allow cross-namespace routes
+	route.Spec.To.Namespace = ""
```

Comment:
I'll do that, but that'll be queued behind the codecgen changes I have. I'd rather get this in now and drop it when I convert to a LocalObjectReference